.eslintrc.js

```diff
@@ -0,0 +1,256 @@
+    "@typescript-eslint/prefer-regexp-exec": "off",
+    "@typescript-eslint/prefer-namespace-keyword": "error",
+    "@typescript-eslint/quotes": ["error", "double"],
+    "@typescript-eslint/restrict-template-expressions": [
+      "error",
+      {
+        allowNumber: true,
+        allowBoolean: true,
+        allowAny: true,
+        allowNullish: false,
+      },
+    ],
+    "@typescript-eslint/semi": ["error"],
+    "@typescript-eslint/triple-slash-reference": [
+      "error",
+      {
+        path: "always",
+        types: "prefer-import",
+        lib: "always",
+      },
+    ],
+    "@typescript-eslint/type-annotation-spacing": "error",
+    "@typescript-eslint/unified-signatures": "error",
+    "arrow-body-style": "off",
+    "arrow-parens": ["off", "always"],
+    "brace-style": ["error", "1tbs"],
+    "comma-dangle": [
+      "error",
+      {
+        objects: "always-multiline",
+        arrays: "always-multiline",
+        functions: "always-multiline",
+        imports: "always-multiline",
+      },
+    ],
+    complexity: "off",
+    "constructor-super": "error",
+    curly: "error",
+    "eol-last": "error",
+    eqeqeq: ["error", "smart"],
+    "guard-for-in": "error",
+    "id-blacklist": [
+      "error",
+      "any",
+      "Number",
+      "number",
+      "String",
+      "string",
+      "Boolean",
+      "boolean",
+      "Undefined",
+      "undefined",
+    ],
+    "id-match": "error",
+    "import/no-default-export": "off",
+    "import/no-unassigned-import": "off",
+    "import/order": ["error", { alphabetize: { order: "asc", caseInsensitive: true } }],
+    "jsdoc/check-alignment": "error",
+    "jsdoc/check-indentation": "error",
+    "jsdoc/newline-after-description": "error",
+    "max-classes-per-file": ["error", 5],
+    // "max-len": [
+    //   "error",
+    //   {
+    //     code: 120,
+    //   },
+    // ],
+    "max-lines": ["error", 700],
+    "new-parens": "error",
+    "no-bitwise": "error",
+    "no-caller": "error",
+    "no-cond-assign": "error",
+    "no-console": "error",
+    "no-debugger": "error",
+    "no-empty": "off",
+    "no-eval": "error",
+    "no-fallthrough": "off",
+    "no-invalid-this": "off",
+    "no-multiple-empty-lines": "error",
+    "no-new-wrappers": "error",
+    "no-null/no-null": "off",
+    "no-restricted-imports": ["error", "rxjs"],
+    "no-shadow": [
+      "error",
+      {
+        hoist: "all",
+      },
+    ],
+    "no-throw-literal": "error",
+    "no-trailing-spaces": "error",
+    "no-undef-init": "error",
+    "no-underscore-dangle": "error",
+    "no-unsafe-finally": "error",
+    "no-unused-labels": "error",
+    "no-var": "error",
+    "object-shorthand": "off",
+    "one-var": ["off", "never"],
+    "padding-line-between-statements": [
+      "off",
+      {
+        blankLine: "always",
+        prev: "*",
+        next: "return",
+      },
+    ],
+    "prefer-arrow/prefer-arrow-functions": "error",
```

Comment:
Not sure about this one: `const demo = () => {}` is 21 character, but `function demo() {}` is 18. Not sure why someone would prefer a longer form.

There is a built-in "[prefer arrow callback](https://eslint.org/docs/rules/prefer-arrow-callback)" rule which makes sense, but using arrow for all is too much imo.

Reply:
it is alignment with our internal code style, so I like to keep it